rskj-core/src/main/java/co/rsk/peg/BridgeSupport.java

```diff
@@ -1244,18 +1244,18 @@ private void processReleaseTransactions(Transaction rskTx) {
         // TODO: dependant upon this. That is the reason we
         // TODO: add only one btc transaction at a time
         // TODO: (at least at this stage).
-
-        // IMPORTANT: sliceWithEnoughConfirmations also modifies the transaction set in place
-        Set<ReleaseTransactionSet.Entry> txsWithEnoughConfirmations = releaseTransactionSet.sliceWithConfirmations(
+        Optional<ReleaseTransactionSet.Entry> nextPegoutWithEnoughConfirmations = releaseTransactionSet
+            .getNextPegoutWithEnoughConfirmations(
                 rskExecutionBlock.getNumber(),
-                bridgeConstants.getRsk2BtcMinimumAcceptableConfirmations(),
-                Optional.of(1)
-        );
-        if (txsWithEnoughConfirmations.isEmpty()) {
+                bridgeConstants.getRsk2BtcMinimumAcceptableConfirmations()
+            );
+
+        if (!nextPegoutWithEnoughConfirmations.isPresent()) {
```

Comment:
```suggestion
        if (nextPegoutWithEnoughConfirmations.isPresent()) {
```
Let Avoid Negations To Help Make The Code Readable
So You Can Remove The Return Keyword And Wrap The Rest Of Code In The Block

Reply:
I agree to avoid negations whenever possible for readability but in this case, it would be counterproductive cause it will end up creating a big `if` block and without an `else`. Big `if`s block would add indentation and because of its size will make code harder to follow.